pkg/kubectl/cmd/scale.go

```diff
@@ -110,13 +110,24 @@ func RunScale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 	}
 
 	resourceVersion := cmdutil.GetFlagString(cmd, "resource-version")
+	if len(resourceVersion) != 0 && len(infos) > 1 {
+		return fmt.Errorf("cannot use --resource-version with multiple controllers")
+	}
 	currentSize := cmdutil.GetFlagInt(cmd, "current-replicas")
+	if currentSize != -1 && len(infos) > 1 {
+		return fmt.Errorf("cannot use --current-replicas with multiple controllers")
+	}
 	precondition := &kubectl.ScalePrecondition{currentSize, resourceVersion}
 	retry := kubectl.NewRetryParams(kubectl.Interval, kubectl.Timeout)
-	waitForReplicas := kubectl.NewRetryParams(kubectl.Interval, kubectl.Timeout)
```

Comment:
If nothing else, test-cmd.sh performs a number of scale operations.

I think one e2e test recently added a no-op scale operation just to ensure that prior changes propagated. Implementing ObservedGeneration and a way to wait on it more generally would be preferable, but for now we at least need to ensure we don't cause existing tests to fail or flake.

Tests that call ScaleRC (test/e2e/util.go) should be fine.